src/logic/safe/transactions/txMonitor.ts

```diff
@@ -0,0 +1,88 @@
+import { TransactionReceipt } from 'web3-core'
+
+import { web3ReadOnly } from 'src/logic/wallets/getWeb3'
+import { sameAddress } from 'src/logic/wallets/ethAddresses'
+import { sameString } from 'src/utils/strings'
+
+type TxMonitorProps = {
+  sender: string
+  hash: string
+  data: string
+  nonce?: number
+  gasPrice?: string
+}
+
+type TxMonitorOptions = {
+  delay?: number
+}
+
+/**
+ * Recursively inspects a pending tx. Until it's found, and returns the mined tx receipt
+ *
+ * @param {object} txParams
+ * @param {string} txParams.sender
+ * @param {string} txParams.hash
+ * @param {string} txParams.data
+ * @param {number | undefined} txParams.nonce
+ * @param {string | undefined} txParams.gasPrice
+ * @param {function(txReceipt: TransactionReceipt): void} cb - called with the tx receipt as argument when tx is mined
+ * @param {object} options
+ * @param {number} options.delay
+ */
+export const txMonitor = async (
+  { sender, hash, data, nonce, gasPrice }: TxMonitorProps,
+  cb: (txReceipt: TransactionReceipt) => void,
+  options?: TxMonitorOptions,
+): Promise<void> => {
+  setTimeout(async () => {
+    if (nonce === undefined || gasPrice === undefined) {
+      // this block is accessed only the first time, to lookup the tx nonce and gasPrice
+      // find the nonce for the current tx
+      const transaction = await web3ReadOnly.eth.getTransaction(hash)
```

Comment:
There are several places like this in txMonitor, where we don't catch a potential async exception.
I think txMonitor should return a promise instead of taking a callback. Then we can try-catch each `await` call and reject the returned promise in the catch.